src/ansible_compat/roles.py

```diff
@@ -0,0 +1,35 @@
+"""Wrapper class for Ansible Roles."""
+import re
+from keyword import iskeyword
+
+
+class Role:
+    """Utility class for validating roles."""
+
+    name_re = re.compile(r"[a-z\d_]+")
+
+    def __init__(self, name: str, check: bool = False):
+        """Instantiate an Ansible role."""
+        self.namespace = ""
+        if name.count(".") == 1:
+            self.namespace, self.name = name.split(".")
+        else:
+            self.name = name
+        if check and (name.count(".") > 1 or not self.validate()):
+            raise RuntimeError(f"Invalid role name: {self}")
+
+    def validate(self) -> bool:
+        """Return false if role is not properly named.
+
+        In the future other tests may be added to validate().
```

Comment:
If we intend to expand the functionality of checks beyond name validation in the future, maybe it would be better to return a list of error messages instead of a boolean? This way consumers can supply users with more details when things will eventually fail to validate.

Reply:
Yeah, i think it will be possible to remove it and keep the API simple. I will do it.

Reply:
I do not want to speculate on what may be needed in the future or not and make the api overly complex. If we will need messaging we can add another `errors()` method that returns False for valid roles or list of errors for invalid ones. 

validate() still expected to be true for valid roles, returning list of errors would still evaluate as true.

@webknjaz WTDY, should I replace `validate()` with `errors()` and reverse its meaning?

Reply:
Validate has no predefined meaning (I would even consider it a bad name for a function that returns boolean). I usually go with `valid?` when programming in Ruby and `is_valid` in Python.

Reply:
I changed implementation based on received feedback, please review again.